### PR TITLE
map: refactor preferences; add thumbnail options

### LINKED ISSUE #8123: FR: map view - temporarily hide images
**Is your feature request related to a problem? Please describe.**
When manually geotagging images by dragging them onto the map, one frequently finds that one (or more) already-placed images obscure the point on the map where the next image should be placed.  This is a problem even when maximally zoomed and using clustering if there are multiple images located within say a block or two of each other.

**Describe the solution you'd like**
It would be nice to have a way of temporarily hiding already-placed images to have an unobstructed view of the map, by (for example) pressing the Alt key while dragging an image.  When the modifier key is released, the hidden images would be redrawn.

**Alternatives**
- Have a switchable mode which shows location pins instead of thumbnails.  That would obscure much less of the map.
- Automatically hide thumbnails within a certain radius of the current mouse location while dragging an image
- Always place images in a strict northeast-to-southwest order :-)

Location pins would be nice to have as an alternate display mode anyway, especially if hovering the mouse on one would show the thumbnail.

(#8106 reminded me to finally post this...)

### PR DESCRIPTION
**Pull request by phweyland**
fix #8123

- map prefs: use preference instropection
- map prefs: add crtl-s (filtered)  and shft-s (thumbnail) as accelerators
- preferences: add ENUM introspection
- map: 3 options for thumbnail display: thumbnail, count of images or nothing
  if "count", hovering it shows the thumbnail, and scroll over it scrolls through the images group
- map: adapt show_OSD API to new prefs

image

hovering or scrolling over a count:
image

### CODE REVIEW

**Comment by phweyland** [comment]
Just solved a memory issue. 
Should be OK for review now. Thanks !

**Comment by Nilvus** [comment]
Just tested and it's really good. I don't make lot of tests but on what I've check, I see no issues... and nothing to do about CSS stuff. With osm-gps-map being now updated and your work also here, map view will become much more interesting. Thanks.

File: src/libs/map_settings.c [hunk 1]
```diff
@@ -199,6 +159,47 @@ void gui_cleanup(dt_lib_module_t *self)
...
+
+  char *str = dt_conf_get_string("plugins/map/images_thumbnail");
+  if(!g_strcmp0(str, "thumbnail"))
+    dt_conf_set_string("plugins/map/images_thumbnail", "count");
+  else if(!g_strcmp0(str, "count"))
+    dt_conf_set_string("plugins/map/images_thumbnail", "none");
+  else
+    dt_conf_set_string("plugins/map/images_thumbnail", "thumbnail");
+  dt_gui_preferences_enum_update(d->images_thumb);
+
```
Comment (TurboGit) [inline src/libs/map_settings.c:186]: you need a g_free(str) here.

File: src/gui/preferences.c [hunk 2]
```diff
@@ -2335,6 +2335,125 @@ GtkWidget *dt_gui_preferences_int(GtkGrid *grid, const char *key)
...
+  gint pos = -1;
+  GList *vals = dt_util_str_to_glist("][", values);
+  for(GList *val = vals; val; val = g_list_next(val))
+  {
+    char *item = (char *)val->data;
+    // remove remaining [ or ]
+    if(item[0] == '[') item++;
+    else if(item[strlen(item) - 1] == ']') item[strlen(item) - 1] = '\0';
+    gtk_list_store_append(store, &iter);
+    // FIXME
```
Comment (TurboGit) [inline src/gui/preferences.c:2427]: not obvious what is to be fixed here, can you add some more comment.

**Review by TurboGit — CHANGES_REQUESTED** [review]
A memory leak to fix. TIA.

File: src/gui/preferences.c (hunk 2 above)
Comment (phweyland) [reply src/gui/preferences.c:2427]: copying bin/preferences_gen.h I was not able to make work:
` C_("preferences", item)`
so I've used 
`_(item)`
which works so far.

Do I explain this or do I remove the comment ?

Comment (TurboGit) [reply src/gui/preferences.c:2427]: Because I suppose content of item is already translated somewhere else but not under the "preferences" context.

Comment (phweyland) [reply src/gui/preferences.c:2427]: Hummm.... My mistake should be elsewhere:

```
preferences.c:2428:68: error: expected ‘)’ before ‘item’
 2428 |     gtk_list_store_set(store, &iter, 0, item, 1, C_("preferences", item), -1);
```

Comment (TurboGit) [reply src/gui/preferences.c:2427]: Meaning that macro` C_()` takes only strings as parameters. I have no solution to propose....

Comment (phweyland) [reply src/gui/preferences.c:2427]: OK. Thank you.

**Comment by phweyland** [comment]
Requested changes done. Thanks for the review.

**Comment by TurboGit** [comment]
Thnaks!

**Comment by ralfbrown** [comment]
Nicely done!  Went and geotagged about 200 photos I had previously not gotten around to tagging using count-only mode, and have no issues to report.

**Pull request merged**